components/SharkPicker.tsx

```diff
@@ -0,0 +1,135 @@
+    desc: "Male great whites on average measure 3.4 to 4.0 m (11 to 13 ft) long, while females at 4.6 to 4.9 m (15 to 16 ft). Adults of this species weigh 522–771 kg (1,151–1,700 lb) on average. Great whites are commonly found in all major oceans.",
+  },
+  {
+    key: "hammerhead",
+    alt: "Hammerhead Shark",
+    src: "/hammerhead.jpg",
+    desc: "The hammerhead sharks are a group of sharks, so named for the unusual structure of their heads, which are flattened and laterally extended into a hammer shape. The known species range from 0.9 to 6.0 m (2 ft 11 in to 19 ft 8 in) in length and weigh from 3 to 580 kg (6.6 to 1,278.7 lb). They are usually light gray and have a greenish tint",
+  },
+  // {
+  //     key: "mako",
+  //     src: { hammerhead },
+  //     desc: ''
+  // },
+  // {
+  //     key: "blue",
+  //     src: { hammerhead },
+  //     desc: ''
+  // },
+  // {
+  //     key: "blue",
+  //     src: { hammerhead },
+  //     desc: ''
+  // },
+  // {
+  //     key: 'soup_fin',
+  //     src: { hammerhead },
+  //     desc: ''
+  // },
+  // {
+  //     key: "six_gill",
+  //     src: { hammerhead },
+  //     desc: ''
+  // },
+];
+
+export const SharkImageCarousel = (
+  shark_id: Shark,
+  left: (() => void) | false,
+  right: (() => void) | false
+) => {
+  return (
+    <VStack alignItems="stretch">
+      <Box>
+        <Image
+          src={shark_id.src}
+          alt={shark_id.alt}
+          width="2500"
+          height="1669"
+        />
+      </Box>
+      <Text fontSize="sm" color="gray">
+        {shark_id.desc}
+      </Text>
+      <HStack>
+        <IconButton
+          aria-label="previous"
+          icon={<ChevronLeftIcon />}
+          isDisabled={!left}
+          onClick={left ? left : undefined}
+          variant="solid"
+        ></IconButton>
+        <Box flexGrow={1} />
+        <IconButton
+          aria-label="next"
+          icon={<ChevronRightIcon />}
+          isDisabled={!right}
+          onClick={right ? right : undefined}
+          variant="solid"
+        ></IconButton>
+      </HStack>
+    </VStack>
+  );
+};
+
+interface SharkPickerProps {
+  shark_index: number | undefined;
+  set_shark_index: (a: number | undefined) => void;
+}
+
+export const SharkPicker: React.FC<SharkPickerProps> = ({
+  shark_index,
+  set_shark_index,
+}) => {
+  // let [shark_index, set_shark_index] = React.useState<number | undefined>();
+  const left =
+    shark_index === 0 || shark_index === undefined
+      ? null
+      : () => set_shark_index(shark_index! - 1);
+  const right =
+    shark_index === sharks.length || shark_index === undefined
+      ? null
+      : () => set_shark_index(shark_index! + 1);
+  //         onSelectionChange={(id) =>
+  //   set_shark_index(sharks.findIndex((obj) => obj.key == id))
+  // }
+  // selectedKey={
+  //   shark_index != undefined ? sharks[shark_index].key : shark_index
+  // }
+  return (
+    <VStack spacing="4">
+      <Field as={Select} label="Shark Type">
```

Comment:
I don't think you should be trying to render a Formik component as a `<Select>`. We probably want to use `Select` directly and set its `value` to the selected shark's name.

I haven't used Chakra's enough to know for sure, but the right move is probably set an `onClick` directly on the `<option>` children themselves. So we should probably generate them dynamically by `map`ping over the shark list and capture the value in the `onClick`.

Check out what `<CheckboxFormField>` does and how they're used- that's probably a good example of how to integrate this with the rest of the form data!